service_crategen/src/commands/generate/codegen/rest_request_generator.rs

```diff
@@ -0,0 +1,263 @@
+
+        if uri_strings.len() > 0 {
+            return Some(format!("let request_uri = format!(\"{request_uri}\", {uri_strings});",
+                                request_uri = generate_snake_case_uri(request_uri),
+                                uri_strings = uri_strings.join(", ")));
+
+        }
+    }
+
+    Some(format!("let request_uri = \"{request_uri}\";",
+                 request_uri = request_uri))
+}
+
+fn generate_shape_member_uri_strings(shape: &Shape) -> Vec<String> {
+    shape.members
+        .as_ref()
+        .unwrap()
+        .iter()
+        .filter_map(|(member_name, member)| {
+            generate_member_format_string(&generate_field_name(member_name), member)
+        })
+        .collect::<Vec<String>>()
+}
+
+fn generate_member_format_string(member_name: &str, member: &Member) -> Option<String> {
+    match member.location {
+        Some(ref x) if x == "uri" => {
+            match member.location_name {
+                Some(ref loc_name) => {
+                    Some(format!(
+                        "{member_name} = input.{field_name}",
+                        field_name = member_name,
+                        member_name = loc_name.to_snake_case(),
+                    ))
+                }
+                None => {
+                    Some(format!(
+                        "{member_name} = input.{field_name}",
+                        field_name = member_name,
+                        member_name = member_name.to_snake_case(),
+                    ))
+                }
+            }
+        }
+        Some(_) | None => None,
+    }
+}
+
+fn generate_static_param_strings(operation: &Operation) -> Vec<String> {
+
+    // botocore includes + for greedy parameters and we don't care about it
+    let (_, maybe_params) = parse_query_string(&operation.http.request_uri);
+
+    let mut static_params = vec![];
+
+    // Sometimes the key has the query param already set, like "list-type=2"
+    // These need to get parsed out of the URI and added to the params map
+    // with the other "queryparam" members
+    if let Some(ref key) = maybe_params {
+        match key.find("=") {
+            Some(_) => {
+                let key_val_vec: Vec<&str> = key.split("=").collect();
+                static_params.push(format!("params.put(\"{}\", \"{}\");",
+                                           key_val_vec[0],
+                                           key_val_vec[1]))
+            }
+            None => static_params.push(format!("params.put_key(\"{}\");", key)),
+        }
+    };
+
+    static_params
+}
+
+fn generate_snake_case_uri(request_uri: &str) -> String {
+    lazy_static! {
+        static ref URI_ARGS_REGEX: Regex = Regex::new(r"\{([\w\d]+)\}").unwrap();
+    }
+
+    URI_ARGS_REGEX.replace_all(request_uri, |caps: &Captures| {
+            format!("{{{}}}",
+                    caps.get(1).map(|c| Inflector::to_snake_case(c.as_str())).unwrap())
```

Comment:
This is a bit confusing. I think what's happening here is, the URL has several params like `{someParam}` in it, and we want to replace that with `{some_param}`? Is that correct?

I'm not super familiar with the regex library, but I believe this would replace all params with just the first capture, so if you had a url like `some/url/{paramA}/{paramB}`, you would get `some/url/{param_a}/{param_a}`.